src/composite.c

```diff
@@ -115,11 +98,93 @@ int Composite(VipsObject *context, VipsImage *src, VipsImage *dst, VipsImage **o
 
   // Combine RGB and alpha channel into output image:
   VipsImage *joined;
-  if (vips_bandjoin2(outRGB, outAlpha, &joined, NULL))
+  if (vips_bandjoin2(outRGBPremultiplied, outAlpha, &joined, NULL))
     return -1;
 
-  // Return a reference to the output image:
-  *out = joined;
+  // Return a reference to the composited output image:
+  *outPremultiplied = joined;
+
+  return 0;
+}
+
+/*
+ * Premultiply alpha channel of `image`.
+ */
+int Premultiply(VipsObject *context, VipsImage *image, VipsImage **out) {
+  VipsImage *imagePremultiplied;
+
+#if (VIPS_MAJOR_VERSION >= 9 || (VIPS_MAJOR_VERSION >= 8 && VIPS_MINOR_VERSION >= 1))
```

Comment:
Will do. After 8.1.0 is out, and you've updated the Heroku buildpack, we can update sharp to rely exclusively on libvips' `vips_*premultiply` methods.